tests/core/selection/optimization.js

```diff
@@ -124,6 +124,19 @@
 
 				assert.isFalse( editor._.shiftPressed );
 			} );
+		},
+
+		// (#3705)
+		'test deleting blocks': function( editor, bot ) {
+			bot.setData( '<p>&nbsp;</p><p>Whatever</p>', function() {
+				var range = editor.createRange();
+
+				range.selectNodeContents( editor.editable() );
+				range.select();
+				editor.extractSelectedHtml();
```

Comment:
Can you also assert what `editor.extractSelectedHtml();` is returning (you can pass `true` as a first argument to get a string)?

Reply:
Safari is the only one that doesn't extract `<p>` element (just unwrapped `<br data-cke-eol="1">`. I tried to track down why this happens and it seems that `range.select()` method sets the `endContainer` to be text node instead of body like in other browsers. For that reason unfortunately I had to distinguish this case in unit test.

Reply:
AFAIR Safari has some native selection optimization mechanism which always places selection boundaries in text nodes (if it can and even if you explicitly place it in block element) so we can't do nothing more reasonable here.